### DISCUSSION TITLE
[Vue warn]: onServerPrefetch is called when there is no active component instance to be associated with.

### DISCUSSION THREAD

**Discussion by mittci** [vuejs/pinia#1797] [Help and Questions] [answered]
### Reproduction

https://github.com/Maxp777/nuxt4

### Steps to reproduce the bug

npm run dev

### Expected behavior

-

### Actual behavior

[Vue warn]: onServerPrefetch is called when there is no active component instance to be associated with. Lifecycle injection APIs can only be used during execution of setup(). If you are using async setup(), make sure to register lifecycle hooks before the first await statement.

### Additional information

with callWithNuxt i get hydration error

```
const update = (): Promise<void> => {
  callWithNuxt(
    nuxtApp,
    listStore.update,
    [{
      element: element.value,
      header: [],
      uri: uri.value
    }]
  )
update()
```

with callWithNuxt and async/await I get the same warning as without callWithNuxt

```
const update = async (): Promise<void> => {
  callWithNuxt(
    nuxtApp,
    await listStore.update,
    [{
      element: element.value,
      header: [],
      uri: uri.value
    }]
  )

await update()
```

**Comment by posva** [comment, marked as answer]
This probably come from the useAsyncData being use within the pinia action instead of being used at the setup level but it’s not coming from pinia itself

**Reply by mittci** [reply]
ok, will try useFetch(), $fetch and $http (@nuxtjs-alt/http)

**Reply by posva** [reply]
note this will happen with composables in general as they’re meant to be called within setup
